pkg/network/encoding/http_test.go

```diff
@@ -340,3 +341,70 @@ func verifyQuantile(t *testing.T, sketch *ddsketch.DDSketch, q float64, expected
 	assert.True(t, val >= expectedValue-acceptableError)
 	assert.True(t, val <= expectedValue+acceptableError)
 }
+
+func generateBenchMarkPayload(sourcePortsMax, destPortsMax uint16) network.Connections {
+	localhost := util.AddressFromString("127.0.0.1")
+
+	payload := network.Connections{
+		BufferedData: network.BufferedData{
+			Conns: make([]network.ConnectionStats, sourcePortsMax*destPortsMax),
+		},
+		HTTP: make(map[http.Key]*http.RequestStats),
+	}
+
+	httpStats1 := http.NewRequestStats(false)
+	httpStats1.AddRequest(100, 10, 0, nil)
+	httpStats1.AddRequest(200, 10, 0, nil)
+	httpStats1.AddRequest(300, 10, 0, nil)
+	httpStats1.AddRequest(400, 10, 0, nil)
+	httpStats1.AddRequest(500, 10, 0, nil)
+
+	for sport := uint16(0); sport < sourcePortsMax; sport++ {
+		for dport := uint16(0); dport < destPortsMax; dport++ {
+			index := sport*sourcePortsMax + dport
+
+			payload.Conns[index].Dest = localhost
+			payload.Conns[index].Source = localhost
+			payload.Conns[index].DPort = dport + 1
+			payload.Conns[index].SPort = sport + 1
+			if index%2 == 0 {
+				payload.Conns[index].IPTranslation = &network.IPTranslation{
+					ReplSrcIP:   localhost,
+					ReplDstIP:   localhost,
+					ReplSrcPort: dport + 1,
+					ReplDstPort: sport + 1,
+				}
+			}
+
+			payload.HTTP[http.NewKey(
+				localhost,
+				localhost,
+				sport+1,
+				dport+1,
+				fmt.Sprintf("/api/%d-%d", sport+1, dport+1),
+				true,
+				http.MethodGet,
+			)] = httpStats1
+		}
+	}
+
+	return payload
+}
+
+func BenchmarkHTTPEncoder100Requests(b *testing.B) {
+	payload := generateBenchMarkPayload(10, 10)
+	b.ResetTimer()
+	b.ReportAllocs()
+	for i := 0; i < b.N; i++ {
+		newHTTPEncoder(&payload)
+	}
+}
+
+func BenchmarkHTTPEncoder10000Requests(b *testing.B) {
```

Comment:
why do you need the BenchmarkHTTPEncoder100Requests and the BenchmarkHTTPEncoder10000Requests as well? can't the functions reuse the first benchmark, in if so and you have to use both of then you can add function in order to avoid code duplication that you have since the creation of the payload to the end